code/code/spec/spec_objs_weapons.cc

```diff
@@ -2524,3 +2524,278 @@ int livingVines(TBeing *vict, cmdTypeT cmd, const char *, TObj *o, TObj *)
   return TRUE;
 }
 
+// Summary of this weapon proc
+//    - 1d8 fire damage proc
+//    - 2% chance of bonebreak, with a fire theme (withers the limb due to burn damage)
+//    - Flavor text proc of dripping lava from the weapon
+int moltenWeapon(TBeing *vict, cmdTypeT cmd, const char *arg, TObj *o, TObj *)
+{
+  char limb[80];
+  int rc, dam = 0, chance;
+  TBeing *ch;
+  sstring buf;
+  wearSlotT slot;
+
+
+  if (!(ch = dynamic_cast<TBeing *>(o->equippedBy)))
+    return FALSE;       // weapon not equipped (carried or on ground)
+
+
+  // Flavor text pulse
+  if (cmd == CMD_GENERIC_PULSE)
+  { 
+    if (!::number(0,75)){
+      act("<r>Drops of molten hot lava fall from $n's $o onto the $g.<1>",
+          0, ch, o, 0, TO_ROOM);
+      act("<r>Your $o drips molten hot lava onto the $g.<1>",
+          0, ch, o, 0, TO_CHAR);
+    }
+  }
+
+
+  if(cmd == CMD_OBJ_HITTING)
+  {
+    chance = ::number(1,50);
+
+
+    if (!o || !vict)
+      return FALSE;
+
+
+    // Bonebreak proc + 10-20 damage if successful (max once/fight)
+    if (chance == 1)
+    {
+       if(!ch->canWither(vict, SILENT_YES))
+         return false;
+
+
+       bool ok = false;
+       bool found = false;
+
+
+
+
+       // Initial checks to prevent issues and for balancee reasons
+       for (slot = MIN_WEAR; slot < MAX_WEAR; slot++) 
+       {
+         if (notBreakSlot(slot, false))  // same ones, right?
+           continue;
+         if (!vict->slotChance(slot))
+           continue;
+         found |= (vict->isLimbFlags(slot, PART_MISSING));
+         ok = true;
+        }
+
+
+
+
+       if (found || !ok) 
+	 return false;
+
+
+       if (vict->isImmune(IMMUNE_HEAT, slot) )
+         return false;
+
+
+
+
+
+
+       for (slot = pickRandomLimb();; slot = pickRandomLimb())  
+       {
+         if (notBreakSlot(slot, true))
+           continue;
+         if (!vict->slotChance(slot))
+           continue;
+         break;
+       }
+       sprintf(limb, "%s", vict->describeBodySlot(slot).c_str());
+
+
+       TThing *t;
+
+
+       if (!vict->hasPart(slot)) {
+         vlogf(LOG_COMBAT, format("BOGUS SLOT trying to be made PART_USELESS: %d on %s") % 
+       	   slot % vict->getName());
+      	      return FALSE;
+       }
+
+
+       if (!vict->roomp) {
```

Comment:
Really feels like 90% of this code should be shared with wither limb prayer

Reply:
A lot of this was originally borrowed from the unmaker proc (which removes limbs altogether), but I moved away from that after discussing with Aion.  Our concern was that missing limbs cannot be healed by players and that can cause some problems from a game balance perspective.  What I would propose is that I come back in a separate branch and rework unmaker/molten and perhaps wither limb as well to pull some of this out into a separate method.  That would let us continue to have different flavor text for different procs/spells without duplicating all the checks, and would allow us to more easily add new procs/spells in the future.

Reply:
Or just pass in the flavor texts as a function parameter, packed into a struct or such.

Reply:
It'd be doable to pass them in.  Generally there's 3 sets of flavor text - to the victim, to the user and to the rest of the room.  Each one can get various parameters, such as the specific limb being hit, although that would likely be determined in the shared function so that'd be fine.  With that said though, I'd want to take a deeper look at these procs to try and put some more thought into it to try and figure out what's the best approach.